x/mint/keeper/keeper_test.go

```diff
@@ -0,0 +1,88 @@
+package keeper_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	sdksimapp "github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/certikfoundation/shentu/v2/simapp"
+)
+
+// shared setup
+type KeeperTestSuite struct {
+	suite.Suite
+	app *simapp.SimApp
+	ctx sdk.Context
+}
+
+func (suite *KeeperTestSuite) SetupTest() {
+	suite.app = simapp.Setup(false)
+	suite.ctx = suite.app.BaseApp.NewContext(false, tmproto.Header{Time: time.Now().UTC()})
+	coins := sdk.Coins{sdk.NewInt64Coin("uctk", 80000*1e6)}
+	suite.Require().NoError(sdksimapp.FundModuleAccount(suite.app.BankKeeper, suite.ctx, "mint", coins))
+}
+
+func (suite *KeeperTestSuite) TestKeeper_SendToCommunityPool() {
+	tests := []struct {
+		name  string
+		coins sdk.Coins
+		err   bool
+	}{
+		{
+			name:  "Funding Community Pool",
+			coins: sdk.NewCoins(sdk.NewInt64Coin(suite.app.StakingKeeper.BondDenom(suite.ctx), 100*1e6)),
+			err:   false,
+		},
+		{
+			name:  "Funding With 0",
+			coins: sdk.NewCoins(sdk.NewInt64Coin(suite.app.StakingKeeper.BondDenom(suite.ctx), 0)),
+			err:   true,
+		},
+	}
+
+	for _, tc := range tests {
+		suite.T().Log(tc.name)
+		err := suite.app.MintKeeper.SendToCommunityPool(suite.ctx, tc.coins)
+		if tc.err {
+			suite.Require().Nil(err)
+		}
+		suite.Require().NoError(err)
+	}
+}
+
+func (suite *KeeperTestSuite) TestKeeper_SendToShieldRewards() {
+	tests := []struct {
+		name  string
+		coins sdk.Coins
+		err   bool
+	}{
+		{
+			name:  "Funding Shield Rewards",
+			coins: sdk.NewCoins(sdk.NewInt64Coin(suite.app.StakingKeeper.BondDenom(suite.ctx), 100*1e6)),
+			err:   false,
+		},
+		{
+			name:  "Funding With 0",
+			coins: sdk.NewCoins(sdk.NewInt64Coin(suite.app.StakingKeeper.BondDenom(suite.ctx), 0)),
+			err:   true,
+		},
+	}
+
+	for _, tc := range tests {
+		suite.T().Log(tc.name)
+		err := suite.app.MintKeeper.SendToShieldRewards(suite.ctx, tc.coins)
```

Comment:
Do you think we can implement a little more elaborate check than just a simple check that there is no error?

Reply:
Sure, Just made a commit, to perform a better check, lemme know your thoughts, on it, if any.

Reply:
Yeah looks great. Thank you 🙏